packages/core/src/tools/mcp-client.ts

```diff
@@ -301,6 +307,44 @@ export async function discoverTools(
   }
 }
 
+const PromptListResponseSchema = z.object({
+  prompts: z.array(
+    z.object({
+      name: z.string(),
+      description: z.string().optional(),
+    }),
+  ),
+});
+
+/**
+ * Discovers and logs prompts from a connected MCP client.
+ * It retrieves prompt declarations from the client and logs their names.
+ *
+ * @param mcpServerName The name of the MCP server.
+ * @param mcpClient The active MCP client instance.
+ */
+export async function discoverPrompts(
+  mcpServerName: string,
+  mcpClient: Client,
+): Promise<void> {
+  try {
+    const response = await mcpClient.request(
+      { method: 'prompts/list', params: {} },
+      PromptListResponseSchema,
+    );
+
+    const promptNames = response.prompts.map((p) => p.name);
+
+    if (promptNames.length > 0) {
+      console.log(
+        `Discovered prompts from ${mcpServerName}: ${promptNames.join(', ')}`,
+      );
```

Comment:
I see so is next step to translate these prompts into slash commands? If so, would it be possible to do that in this PR?

I know that expands the size of this PR buttttt I think it may be required to ensure this represents a full unit of work